src/cargo/ops/cargo_compile.rs

```diff
@@ -251,12 +253,18 @@ fn scrape_target_config(config: &Config, triple: &str)
             let key = format!("{}.{}", key, k);
             let (v, path) = try!(config.get_string(&key)).unwrap();
             if k == "rustc-flags" {
+                let _ = config.shell().warn("rustc-flags is deprecated, use
+                            rustc-link-lib and rustc-link-search instead");
```

Comment:
I'm not sure that we want to issue warnings just yet, and it's unclear whether we want to issue warnings at all in general. I'll write up a bigger comment on the PR.